esmvalcore/_recipe.py

```diff
@@ -1252,6 +1252,13 @@ def initialize_tasks(self):
 
             # Create preprocessor tasks
             for variable_group in diagnostic['preprocessor_output']:
+                if not diagnostic['preprocessor_output'][variable_group]:
```

Comment:
It's very late in the process of reading the recipe to put this here, a better place would be to do it here:
https://github.com/ESMValGroup/ESMValCore/blob/e1a4a117cb4230b3a4751ec09f6e89fa057e887a/esmvalcore/_recipe.py#L1013
maybe just change the check from checking only for duplicate datasets to also checking for no datasets at all

Reply:
what a good point! cheers dude!

Reply:
OK changed it and had to remind meself of the exception handling with Fire :fire: